pkgs/applications/finance/odoo/odoo16.nix

```diff
@@ -0,0 +1,115 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchzip
+, python310
+, rtlcss
+, wkhtmltopdf
+, nixosTests
+}:
+
+let
+  python = python310.override {
+    packageOverrides = self: super: {
+      flask = super.flask.overridePythonAttrs (old: rec {
+        version = "2.3.3";
+        src = old.src.override {
+          inherit version;
+          hash = "sha256-CcNHqSqn/0qOfzIGeV8w2CZlS684uHPQdEzVccpgnvw=";
+        };
+      });
+      werkzeug = super.werkzeug.overridePythonAttrs (old: rec {
+        version = "2.3.7";
+        src = old.src.override {
+          inherit version;
+          hash = "sha256-K4wORHtLnbzIXdl7butNy69si2w74L1lTiVVPgohV9g=";
+        };
+      });
+    };
+  };
+
+  odoo_version = "16.0";
+  odoo_release = "20231024";
+in python.pkgs.buildPythonApplication rec {
+  pname = "odoo";
+  version = "${odoo_version}.${odoo_release}";
+
+  format = "setuptools";
+
+  # latest release is at https://github.com/odoo/docker/blob/master/16.0/Dockerfile
+  src = fetchzip {
+    url = "https://nightly.odoo.com/${odoo_version}/nightly/src/odoo_${version}.zip";
+    name = "${pname}-${version}";
+    hash = "sha256-Ux8RfA7kWLKissBBY5wrfL+aKKw++5BxjP3Vw0JAOsk="; # odoo
+  };
+
+  # needs some investigation
+  doCheck = false;
+
+  makeWrapperArgs = [
+    "--prefix" "PATH" ":" "${lib.makeBinPath [ wkhtmltopdf rtlcss ]}"
```

Comment:
[`wkhtmltopdf` depends on `qtwebkit`, which is long unmaintained and marked insecure.](https://github.com/NixOS/nixpkgs/issues/297646) Is it absolutely certain oboo still needs this? Because at the moment, Nixpkgs refuses to evaluate it without enabling insecure packages

Reply:
Good question! I'll look into this.

Reply:
I've gone through the source and the news is not good here: `wkhtmltopdf` is used by builtin `i18n` and `ir` modules:

- it's still in use
- it can't be made an optional feature